analyzers/src/SonarAnalyzer.CSharp/Rules/SymbolicExecution/InitializationVectorShouldBeRandom.cs

```diff
@@ -0,0 +1,248 @@
+                explodedGraph.AddExplodedGraphCheck(new InitializationVectorCheck(explodedGraph, this));
+
+            public IEnumerable<Diagnostic> GetDiagnostics() =>
+                locations.Distinct().Select(location => Diagnostic.Create(Rule, location));
+
+            public void AddLocation(Location location) => locations.Add(location);
+
+            public void Dispose()
+            {
+                // Nothing to do here.
+            }
+        }
+
+        private sealed class InitializationVectorCheck : ExplodedGraphCheck
+        {
+            private static readonly ImmutableArray<KnownType> VulnerableTypes =
+                ImmutableArray.Create(KnownType.System_Security_Cryptography_Aes,
+                                      KnownType.System_Security_Cryptography_AesCng,
+                                      KnownType.System_Security_Cryptography_AesCryptoServiceProvider,
+                                      KnownType.System_Security_Cryptography_SymmetricAlgorithm);
+
+            private readonly AnalysisContext context;
+
+            public InitializationVectorCheck(AbstractExplodedGraph explodedGraph, AnalysisContext context) : base(explodedGraph) =>
+                this.context = context;
+
+            public override ProgramState PostProcessInstruction(ProgramPoint programPoint, ProgramState programState)
+            {
+                programState = programPoint.CurrentInstruction switch
+                {
+                    ObjectCreationExpressionSyntax objectCreation => ObjectCreationPostProcess(objectCreation, programState),
+                    InvocationExpressionSyntax invocation => InvocationExpressionPostProcess(invocation, programState),
+                    ArrayCreationExpressionSyntax arrayCreation => ArrayCreationPostProcess(arrayCreation, programState),
+                    AssignmentExpressionSyntax assignment => AssignmentExpressionPostProcess(assignment, programState),
+                    _ => programState
+                };
+
+                return base.PostProcessInstruction(programPoint, programState);
+            }
+
+            public override ProgramState PreProcessInstruction(ProgramPoint programPoint, ProgramState programState)
+            {
+                if (programPoint.CurrentInstruction is InvocationExpressionSyntax invocation)
+                {
+                    programState = InvocationExpressionPreProcess(invocation, programState);
+                }
+
+                return base.PreProcessInstruction(programPoint, programState);
+            }
+
+            private ProgramState ObjectCreationPostProcess(ObjectCreationExpressionSyntax objectCreation, ProgramState programState)
+            {
+                if (semanticModel.GetSymbolInfo(objectCreation).Symbol is {} symbol
+                    && symbol.ContainingType is {} symbolType
+                    && symbolType.IsAny(VulnerableTypes))
```

Comment:
Can be simplified, `IsAny` checks for null.
```suggestion
                    && symbol.ContainingType.IsAny(VulnerableTypes))
```

Reply:
Does not apply after the latest changes.